### PR TITLE
Fix javadoc errors and add explicit constructors

### PR DESCRIPTION
**Pull request by keithc-ca**
Fix javadoc errors seen in https://openj9-jenkins.osuosl.org/job/Build_JDKnext_aarch64_linux_Personal/47.

Also
* tidy up

### CODE REVIEW

File: jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java [hunk 1]
```diff
@@ -28,32 +28,37 @@
...
+ * Factory for creating different types of contexts.
  *
+ * @author adam
  */
 public class ContextFactory {
-	
+
+	private ContextFactory() {
+		/* no instances needed */
+	}
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java:39]: This class was internal until #17064; the class only has static features so instances are not required.

**Comment by keithc-ca** [comment]
Testing underway in https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/572; see https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJDKnext/140.

File: jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java [hunk 2]
```diff
@@ -28,32 +28,37 @@
...
 	 * @param space address space for this context (cannot be null)
-	 * @param process in this address space
-	 * @param rt Java runtime for this context (may be null)
+	 * @param process process in this address space
+	 * @param runtime Java runtime for this context (may be null)
 	 * @return the context
 	 */
-	public static IDTFJContext getContext(final int major, final int minor,final Image image, final ImageAddressSpace space, final ImageProcess proc, final JavaRuntime rt) {
-		DTFJContext ctx = new DTFJContext(major, minor, image, space, proc, rt);
+	public static IDTFJContext getContext(final int major, final int minor,final Image image, final ImageAddressSpace space, final ImageProcess process, final JavaRuntime runtime) {
```
Comment (JasonFengJ9) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java:51]: `@param image` is missing.

Comment (keithc-ca) [reply jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java:51]: Thanks. I'll fix that.

Comment (keithc-ca) [reply jcl/src/openj9.dtfj/share/classes/com/ibm/java/diagnostics/utils/ContextFactory.java]: Fixed in https://github.com/eclipse-openj9/openj9/pull/17077/commits/05fa5024114cb40275bd54d22224e85bcf21467d.

**Review by JasonFengJ9 — APPROVED** [review]
LGTM

**Comment by pshipton** [comment]
Pls fix the commit title to indicate constructors are also updated.

**Comment by keithc-ca** [comment]
Now the commit title and the headline here mention the constructors.

**Review by pshipton — APPROVED** [review]

**Comment by pshipton** [comment]
Pls cherry pick for 0.38

**Pull request merged**